projects/packages/videopress/src/client/block-editor/blocks/video/components/player/controls/index.native.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon } from '@wordpress/components';
+import { useState, useEffect, useCallback, useRef } from '@wordpress/element';
+/**
+ * External dependencies
+ */
+import { View, Pressable } from 'react-native';
+/**
+ * Internal dependencies
+ */
+import PauseIcon from './icons/icon-pause.native.js';
+import PlayIcon from './icons/icon-play.native.js';
+import ReplayIcon from './icons/icon-replay.native.js';
+import style from './style.scss';
+
+const PlayerControls = ( { isSelected, playEnded, onToggle } ) => {
```

Comment:
When I first started working on this, the `Player` was still a small component. It ballooned a bit more after I added the preview logic. Since the player controls are 1. Android only and 2. hopefully a temporary work around I opted to put as much of the logic into a separate component.